pinot-core/src/main/java/org/apache/pinot/core/transport/QueryServer.java

```diff
@@ -110,7 +119,9 @@ private void attachSSLHandler(SocketChannel ch) {
         throw new IllegalArgumentException("Must provide key store path for secured server");
       }
 
-      SslContextBuilder sslContextBuilder = SslContextBuilder.forServer(TlsUtils.createKeyManagerFactory(_tlsConfig));
+      SslContextBuilder sslContextBuilder = SslContextBuilder
+          .forServer(TlsUtils.createKeyManagerFactory(_tlsConfig))
+          .sslProvider(SslProvider.OPENSSL);
```

Comment:
Agreed on the idea of adding native implementation to TLsConfig. Will do.
For creating a streamlined creation method in TlsUtils I think we can start another PR dedicated for this inssue, so that this PR can be on encryption and authz.

Reply:
done